### ISSUE TITLE
UR decode receive part fails because of crc32 checksum calculation

### ISSUE THREAD

**Issue by newtonick** [Foundation-Devices/foundation-ur-py#3]
Occasionally UR decoding fails to decode a fragment when it shouldn't. It fails to decode the fragment because the checksum included in the body does not match the calculated checksum. I believe the issue is that the calculated checksum is the wrong byte size. Here is an example fragment that reproduces the issue. Other UR libs appear to decode this fragment successfully.

```
fragment = "UR:CRYPTO-PSBT/20-29/LPBBCSCACFCMCPCYBBRSPTSKHDSSDTSBTKDECHRHPKHKVWMDMNKSGDAOAEAEAEAEAECHPTBBEODKPLETLBLDAMJOPMBNPEPLWMFZLTZTHGOEQZTELTLGAYCHAEAEAEAEAECHPTBBFXSAWKIDLTENBSKPJLFEFLMNCLPKEMTPWPMHMHIOLTKECSAMAEAEAEAEAECHPTBBINTKMWZMYNKNKEZCBGKTHDFEZOPAYNPRMEFTHPKGLTAEAEAEAEADADDNEMMDAAAEAEAEAEAECPAECXHDGWFDVTSPHDOLBNKIGETEECLKOSOXLPJSSNFXSGCLAHESJSVTURDYJZCWSRKNDTADAHTKGHCLAOFSTDLNUYSAASIESFDNRKHSMNJZTONLPSLDWFTDMNINOXEHHNKOTODRWPCHRORHDACLAEETWDVL"

ur_decoder = URDecoder()
status = ur_decoder.receive_part(fragment)
````

Bytewords.decode throws the exception 'Invalid Bytewords.' because the calculated crc32 checksum does not match the checksum on the body of the fragment. I believe the issue is that the calculated checksum is not returning from `crc32_bytes` as a 32 bit integer.

For the above example the checksum included in the fragment is `0038eae3` in hex. But the checksum returned by `crc32_bytes` is `38eae3`. So on line 110 of bytewords.py the check `if checksum != body_checksum:` is false.

When I hard code the byte length in the `crc32n` function to be 4 bytes this seems to fix the issue, but I don't know what the "correct" way would be to handle this.

I changed:
`return n.to_bytes((bit_length(n) + 7) // 8, 'big')`

to:
`return n.to_bytes(4, 'big')`

and this makes this issue go away as far as I can tell

**Comment by zachherbert** [comment]
Thanks for reporting @newtonick! 

@FoundationKen want to make sure you see this.

**Comment by FoundationKen** [comment]
Hi @newtonick!  Sorry, I missed the notification for this issue.

I checked the code, and `crc32n()` really should always return 4 bytes, even if there are leading zero bytes, so actually, I think your proposed fix is the right thing to do.  Thanks for digging in on this.

Would you like to submit a PR with the above change?  If not, we'll get this fixed up in the next day or two.

**Comment by FoundationKen** [comment]
Fixed in [56c423c](https://github.com/Foundation-Devices/foundation-ur-py/commit/56c423cbb531b4f5f1625df456addcf15cccd05c).

**Comment by newtonick** [comment]
Thank you for fixing!
